Update multiple-focus-metavariables.md (#745)
returntocorp/semgrep-docs: docs/experiments/multiple-focus-metavariables.md

description: "With this rule, Semgrep matches all pieces of code captured by the focus metavariables."
---

# Including multiple focus metavariables using set union semantics

Semgrep matches all pieces of code captured by focus metavariables when you specify them in a rule. Specify the metavariables you want to focus on in a YAML list format.

:::info
This feature is using `focus-metavariable`, see [`focus-metavariable`](/writing-rules/rule-syntax/#focus-metavariable) documentation for more information.
:::

There are two ways in which you can include multiple focus metavariables:

> - **Set union**: Experimental feature described below in the section [Set union](#set-union). This syntax returns the union of all matches of the specified metavariables.

Author: ```suggestion
- **Set union**: Experimental feature described below in the section [Set union](#set-union). This feature returns the union of all matches of the specified metavariables.
```

# Using multiple focus metavariables using set union semantics

Semgrep matches all pieces of code captured by focus metavariables when you specify them in a rule. Specify the metavariables you want to focus on in a YAML list format.

:::info
This feature is using `focus-metavariable`, see [`focus-metavariable`](/writing-rules/rule-syntax/#focus-metavariable) documentation for more information.
:::

There are two ways in which you can include multiple focus metavariables:

> - **Set union**: Experimental feature described below in the section [Set union](#set-union). This syntax matches metavariables regardless of their position in the code.

Reviewer: I am not sure a user would understand what "This syntax matches metavariables regardless of their position in the code" really means. I am not sure I understand it myself. What this feature does is to return the union of all matches of the specified metavariables.

Author: ```suggestion
- **Set union**: Experimental feature described below in the section [Set union](#set-union). This syntax returns the union of all matches of the specified metavariables.
```
